lib/codecs/src/decoding/framing/mod.rs

```diff
@@ -34,23 +36,33 @@ use super::StreamDecodingError;
 /// It requires conformance to `TcpError` so that we can determine whether the
 /// error is recoverable or if trying to continue will lead to hanging up the
 /// TCP source indefinitely.
-pub trait FramingError: std::error::Error + StreamDecodingError + Send + Sync {}
+pub trait FramingError: std::error::Error + StreamDecodingError + Send + Sync + Any {
+    /// Coerces the error to a `dyn Any`.
+    /// This is useful for downcasting the error to a concrete type, as we are dealing
+    /// with Box<dyn FramingError> instead of a concrete `FramingError` enum.
+    fn as_any(&self) -> &dyn Any;
```

Comment:
I had to do this "hack" in order to downcast a `Box<dyn FramingError>` into a concrete `ChunkedGelfDecoderError` in the `chunked_gelf` unit tests.. This is a bit ugly and I don't know if it have sense to change the `BoxedFramingError` into a concrete `FramingError` enum which would have three variants; `std::io::Error`, `LinesCodecError` and `ChunkedGelfDecoderError`... What do you think? Is there a `Box<dyn FramingError>` for a specific reason? using an enum instead would make the tests I added in `decoding::framing::chunked_gelf::tests`simpler and we would avoid vtables

It shouldn't be a big change to do that. I highly prefer to use concrete enum types instead of `Box<dyn Error>`, at least on lib crates

Reply:
I feel like @bruceg might have thoughts here. Is there a way we could just do implement this in the `chunked_gelf` tests? It seems like this is the first framing method where the tests needed to downcast.